### PR TITLE
Create an interactive, smart and more user-friendly CLI

### LINKED ISSUE #40: Prompt user to add more configuration to the tool
**Is your feature request related to a problem? Please describe.**
As of now, the user just types the command and has no interaction with the CLI tool. 

**Describe the solution you'd like**
It could be cool if he were prompted for what he really wants to do and so I htink it would be more user friendly.

A good application of this feature would be to ask him, the first time he uses `reacli`, if he wants to create a `.reacli` to add his configuration.

Here is inspiration material:

prompting

Linked to: #39

### PR DESCRIPTION
**Pull request by gdarchen**
# Fixes #40 and #44 

**Summary:**

The user previously had to type each option he wanted to use to configure the component to create. I added here a feature enabling the user to only execute `reacli` and the tool asks him stuff so as to get enough information on what thing to create, where to create it and with which options.

I also found the CLI quite boring and ugly. Hence, I decided to add a `figlet` writing "Reacli" when using the tool. I also added `chalk` to each `console.log()` calls so as to make them colorful 😄 

You can see an example on the following picture: 

image

[EDIT] The interactive CLI now detects the `.reacli` config file. It then pre-checks the options to choose when creating a component (which fixes issue #44). 

**Changes:**

* Add an interactive CLI with `inquirer`
* Reorganize some methods in files
* Add a ASCIIart with `figlet`
* Add colors in `console.log()` calls with `chalk`

**Dependencies:**

No dependency

**Instructions:**

Do not forget to run `npm run install` to make sure you download the new packages to install.

### CODE REVIEW

**Review by lcor-io — COMMENTED** [review]
Perfect for me. I noticed several points tho:
- Interactive CLI doesn't work with `git bash`
- Instead of asking for the component's path, I think it could be better to ask for the component's name instead, as making subdirectories doesn't work yet anyway

**Comment by gdarchen** [comment]
> Perfect for me. I noticed several points tho:
> 
> * Interactive CLI doesn't work with `git bash`
> * Instead of asking for the component's path, I think it could be better to ask for the component's name instead, as making subdirectories doesn't work yet anyway

You are right! You can see `inquirer` supported terminals [here](https://www.npmjs.com/package/inquirer#support).

For the second point, I am not sure it's a good idea not to ask the path even if it doesn't work yet. Indeed, we will be solving this point really soon and want to be able to create directories recursively. Our target is then to know the path where to create and not just its name, isn't it?

**Comment by lcor-io** [comment]
> > Perfect for me. I noticed several points tho:
> > 
> > * Interactive CLI doesn't work with `git bash`
> > * Instead of asking for the component's path, I think it could be better to ask for the component's name instead, as making subdirectories doesn't work yet anyway
> 
> You are right! You can see `inquirer` supported terminals [here](https://www.npmjs.com/package/inquirer#support).
> 
> For the second point, I am not sure it's a good idea not to ask the path even if it doesn't work yet. Indeed, we will be solving this point really soon and want to be able to create directories recursively. Our target is then to know the path where to create and not just its name, isn't it?

Yes you're right, but still while I was using the interactive CLI, I instinctively typed `.` for the path, because I was expecting it to ask for the name of the component right after. It resulted in the CLI trying to create a `reacli` folder, obviously not what I was expecting. Maybe we could change the function to validate the path to avoid that ?

**Comment by gdarchen** [comment]
Alright I misunderstood your point 😉 maybe we could just change the CLI sentence asking to enter the path? 
Or maybe we should do as you said but I guess it would be in another issue, isn't it ?

**Comment by lcor-io** [comment]
No problem. Just changing the sentence is enough for now I think, if the problem persist for other users we can still open another issue at that time.
What do you think of something like:  `Enter the component's path (components will have the name of the last folder selected): `

**Comment by gdarchen** [comment]
It should be better now 😊

**Comment by gdarchen** [comment]
It should be mergeable now. Please do not forget to test it since I had to change a lot of things 😉

**Comment by tanohzana** [comment]
- [x] Feature is functional
- [ ] Tests pass
- [ ] Code is clean

File: bin/reacli.js [hunk 1]
```diff
@@ -21,11 +29,16 @@ const reactCli = async () => {
 		.option("--redux", "Add Redux to the template")
 		.parse(process.argv)
 
+	let options = {}
 	const { args } = program
+
+	if (!args.length) {
+		options = await interactiveCLI(options)
```
Comment (tanohzana) [inline bin/reacli.js]: Why do you pass `options` as a param if it is empty ?

File: lib/interactiveCLI.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			choices: ["Component", "Hook"],
+			message: "What would you like to create?",
+			name: "type",
+			type: "list",
+		},
+		{
+			message: "Enter the component's path (components will take their name from the last folder of the path).",
+			name: "path",
+			type: "input",
+			when: (answers) => answers.type === "Component",
```
Comment (tanohzana) [inline lib/interactiveCLI.js:30]: Hook feature is now merged. Would you mind implementing the hook part also ?

File: lib/interactiveCLI.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+			return acc
+		}, {})
+	}
+
+	// Create the component
+	await createComponent(pathModule.resolve(answers.path), options)
+
+	if (answers.type === "Hook") {
+		console.log(chalk.black.bgRed("Not implemented yet."))
```
Comment (tanohzana) [inline lib/interactiveCLI.js]: Implemented now

File: lib/interactiveCLI.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	}
+
+	return options
+}
+
+export { interactiveCLI }
```
Comment (tanohzana) [inline lib/interactiveCLI.js]: Prefer `export default interactiveCLI` if it's the only thing that's exported

**Review by tanohzana — CHANGES_REQUESTED** [review]
A few things to check

File: bin/reacli.js (hunk 1 above)
Comment (gdarchen) [reply bin/reacli.js]: Should be fixed by https://github.com/reacli/cli/pull/43/commits/6b5935ee02304b43727d0f9f875258ee90999185

File: lib/interactiveCLI.js (hunk 4 above)
Comment (gdarchen) [reply lib/interactiveCLI.js]: https://github.com/reacli/cli/pull/43/commits/c774fdd5fbcd95e93807f8675b89822e69f061a5 should solve it 😊

File: lib/interactiveCLI.js (hunk 2 above)
Comment (gdarchen) [reply lib/interactiveCLI.js:30]: It should be ok with https://github.com/reacli/cli/pull/43/commits/ec782719db342247d5379b2162c9513342a0fa75.

**Comment by gdarchen** [comment]
Ok, I solved a loooot of conflicts and integrated the hook feature into the interactive CLI.
Since we still do not have tests, I would really recommend you to test again that everything is working properly, what I already did but...never too sure 😊

**Review by tanohzana — APPROVED** [review]

**Comment by tanohzana** [comment]
Great job 😁

**Pull request merged**